warplib/dynamic_arrays_util.cairo

```diff
@@ -231,3 +231,32 @@ func bytes_upper_bound{range_check_ptr: felt}(number: felt) -> (upper_bound: fel
     let (result) = bytes_upper_bound(number - 32);
     return (32 + result,);
 }
+
+// Utility functions for indexed events
+
+// Memory felt array to felt array copy
+// ---- mem_loc: warp memory location of the array
+// ---- array: felt array to write to
+// ---- index : the location at which to start writing to array
+func memory_array_to_felt_array{range_check_ptr: felt, warp_memory: DictAccess*}(
```

Comment:
https://github.com/NethermindEth/warp/blob/develop/warplib/memory.cairo#L357
we have an implementation of this function